### PR TITLE
Expose TRAVIS_JOB_NAME

### PR DESCRIPTION
**Pull request by backspace**

### CODE REVIEW

**Comment by backspace** [comment]
You can see this working [here](https://staging.travis-ci.org/backspace/travixperiments/jobs/731889#L377). It works in concert with [this `travis-scheduler`](https://github.com/travis-ci/travis-scheduler/pull/178) PR.

File: lib/travis/build/env/builtin.rb [hunk 1]
```diff
@@ -53,6 +53,7 @@ def env_vars
               TRAVIS_PULL_REQUEST_BRANCH: job[:pull_request_head_branch],
               TRAVIS_PULL_REQUEST_SHA: job[:pull_request_head_sha],
               TRAVIS_PULL_REQUEST_SLUG: job[:pull_request_head_slug],
+              TRAVIS_JOB_NAME:        job[:name]
```
Comment (BanzaiMan) [inline lib/travis/build/env/builtin.rb]: It's a small thing, but adding a trailing `,` helps to minimize the next PR.

Also, while there is no order in `TRAVIS_*` env vars, it seems a good idea to keep this one close to other `TRAVIS_JOB_*` env vars.

**Comment by BanzaiMan** [comment]
Please test with special characters in the job name; spaces, parentheses, dollar signs, etc.

**Comment by backspace** [comment]
Thanks for the suggestion, [here](https://staging.travis-ci.org/backspace/travixperiments/jobs/731894#L377) is an example.

**Pull request merged**